streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsAssignmentScaleTest.java

```diff
@@ -0,0 +1,256 @@
+            partitionInfos,
+            emptySet(),
+            emptySet()
+        );
+
+        final InternalTopologyBuilder builder = new InternalTopologyBuilder();
+        builder.addSource(null, "source", null, null, null, "topic");
+        builder.addProcessor("processor", new MockApiProcessorSupplier<>(), "source");
+        builder.addStateStore(new MockKeyValueStoreBuilder("store", false), "processor");
+        builder.setApplicationId(APPLICATION_ID);
+        builder.buildTopology();
+
+        final Consumer<byte[], byte[]> mainConsumer = EasyMock.createNiceMock(Consumer.class);
+        final TaskManager taskManager = EasyMock.createNiceMock(TaskManager.class);
+        expect(taskManager.builder()).andStubReturn(builder);
+        expect(taskManager.mainConsumer()).andStubReturn(mainConsumer);
+        expect(mainConsumer.committed(new HashSet<>())).andStubReturn(Collections.emptyMap());
+        final AdminClient adminClient = createMockAdminClientForAssignor(changelogEndOffsets);
+
+        final StreamsPartitionAssignor partitionAssignor = new StreamsPartitionAssignor();
+
+        final Map<String, Object> configMap = new HashMap<>();
+        configMap.put(StreamsConfig.APPLICATION_ID_CONFIG, APPLICATION_ID);
+        configMap.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:8080");
+        configMap.put(InternalConfig.TASK_MANAGER_FOR_PARTITION_ASSIGNOR, taskManager);
+        configMap.put(InternalConfig.STREAMS_METADATA_STATE_FOR_PARTITION_ASSIGNOR, EasyMock.createNiceMock(StreamsMetadataState.class));
+        configMap.put(InternalConfig.STREAMS_ADMIN_CLIENT, adminClient);
+        configMap.put(InternalConfig.ASSIGNMENT_ERROR_CODE, new AtomicInteger());
+        configMap.put(InternalConfig.NEXT_SCHEDULED_REBALANCE_MS, new AtomicLong(Long.MAX_VALUE));
+        configMap.put(InternalConfig.TIME, new MockTime());
+        configMap.put(InternalConfig.INTERNAL_TASK_ASSIGNOR_CLASS, taskAssignor.getName());
+        configMap.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, numStandbys);
+
+        final MockInternalTopicManager mockInternalTopicManager = new MockInternalTopicManager(
+            new MockTime(),
+            new StreamsConfig(configMap),
+            new MockClientSupplier().restoreConsumer,
+            false
+        );
+        partitionAssignor.configure(configMap);
+        EasyMock.replay(taskManager, adminClient, mainConsumer);
+
+        partitionAssignor.setInternalTopicManager(mockInternalTopicManager);
+
+        final Map<String, Subscription> subscriptions = new HashMap<>();
+        for (int client = 0; client < numClients; ++client) {
+            for (int i = 0; i < numThreadsPerClient; ++i) {
+                subscriptions.put(getConsumerName(i, client),
+                                  new Subscription(
+                                      topic,
+                                      getInfo(uuidForInt(client), EMPTY_TASKS, EMPTY_TASKS).encode())
+                );
```

Comment:
Code style suggestion:
```suggestion
                subscriptions.put(
                    getConsumerName(i, client),
                    new Subscription(topic, getInfo(uuidForInt(client), EMPTY_TASKS, EMPTY_TASKS).encode())
                );
```